exp/internal/controllers/machinepool_controller_noderef_test.go

```diff
@@ -181,7 +182,7 @@ func TestMachinePoolGetNodeReference(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			result, err := r.getNodeReferences(ctx, client, test.providerIDList)
+			result, err := r.getNodeReferences(ctx, client, test.providerIDList, ptr.To[int32](0))
```

Comment:
I think it would help to add a unit test case that tests the change in behavior for `available`. It doesn't seem like any of the existing test cases check for `available` besides the empty case.

Reply:
lgtm otherwise!